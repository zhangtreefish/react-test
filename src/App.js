import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import logo from './logo.svg';
import './App.css';

export class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

App.propTypes = {
  errors: PropTypes.array,
};

// mapStateToProps :: {State} -> {Props}
const mapStateToProps = (state) => ({
  errors: state.App.errors,
});

export default connect(mapStateToProps)(App);
