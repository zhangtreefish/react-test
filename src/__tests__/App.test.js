import React from 'react';
import ReactDOM from 'react-dom';
import { App } from '../App';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';

it('renders without crashing', () => {
  // const div = document.createElement('div');
  // ReactDOM.render(<App />, div);
  var tree = renderer.create(<App redux={fromJS({})} />).toJSON();
  expect(tree).toMatchSnapshot();
});
