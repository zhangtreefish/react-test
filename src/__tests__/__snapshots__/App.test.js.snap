exports[`test renders without crashing 1`] = `
<div
  className="App">
  <div
    className="App-header">
    <img
      alt="logo"
      className="App-logo"
      src="test-file-stub" />
    <h2>
      Welcome to React
    </h2>
  </div>
  <p
    className="App-intro">
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
